src/k8s/cmd/k8s/k8s_status.go

```diff
@@ -27,16 +27,6 @@ func newStatusCmd(env cmdutil.ExecutionEnvironment) *cobra.Command {
 				return
 			}
 
-			// TODO(neoaggelos): this must be done on the server side
-			// fail fast if we're not explicitly waiting and we can't get kube-apiserver endpoints
-			if !opts.waitReady {
-				if !client.IsKubernetesAPIServerReady(cmd.Context()) {
```

Comment:
Please also remove the implementation of `client.IsKubernetesAPIServerReady()` as it not used anywhere anymore (remove from client implementation and the interface)